### DISCUSSION TITLE
Status for thunderbolt support?

### DISCUSSION THREAD

**Discussion by sk0n3** [hackintosh-club/intel-nuc10#68] [General]
The description said very buggy but my thunderbolt doesn't work at all right now and I was wondering if everybody is on the same page on that.
It's registered as USB 3.1 device

**Comment by ghost** [comment]
![thunderbolt JHL7540](https://user-images.githubusercontent.com/96088484/177466429-0a7e3201-dc00-4f69-9a36-36df2447ea3c.png)

I haven't tested it however this may be of help. The thunderbolt 3 controller in NUC10 is Intel JHL7540.

>https://elitemacx86.com/threads/how-to-enable-thunderbolt-3-hotplug-on-macos.462/
> Implementation of Hotplug with Thunderbolt has been always a major issue with macOS. But Thanks to users like KGP who have been able to implement it on the hardware. The reason for the not working of Hotplug with the Thunderbolt is due to the working style of thunderbolt on both the platform i.e. Windows and Mac. On Windows hardware, the working of Thunderbolt is different as compared to a real Macintosh.
> 
> Before getting into it, let's get to know about the working of ThunderBolt on Mac and Windows hardware which will help you to understand the working of Hotplug.
> 
> On Windows hardware, the Thunderbolt is only powered up when any Thunderbolt compatible Device is connected. The device triggers the thunderbolt to function and turns on the power to the connected device. Once, it's disconnected, the power to the port is also turned off.
> 
> On the other hand, on Mac, it's not the same. On Mac, the Thunderbolt is always on until the system is turned off. When any device is connected to the Thunderbolt port, the Device automatically connects to the thunderbolt and this is the reason the hotplug basically works. The Thunderbolt is always active as it keeps drawing the power.
> 
> On the Hackintosh, it's a little bit different or you can say, it's mixed. The thunderbolt will function only if the device is connected before booting to macOS i.e preboot mode. If you disconnect it, the device no longer functions and you need to restart the system again. Hence, the system is not capable of Hotplug.
